File: LeetCode/Tricks/SlidingWindow/Js/Leetcode713-乘积小于K的子数组.js
var numSubarrayProductLessThanK = function (nums, k) {
    let [acc, res, n] = [1, 0, nums.length];
    for (let i = 0, j = 0; j < n; j++) {
        acc *= nums[j];
        while (i <= j && acc >= k) {
            acc = Math.floor(acc / nums[i]);
            i++;
        }
        res += j - i + 1;
    }
    return res;
};

File: LeetCode/Tricks/SlidingWindow/Go/Leetcode713-乘积小于K的子数组.go
package sbgo

func numSubarrayProductLessThanK(nums []int, k int) int {
	res, acc, n := 0, 1, len(nums)
	for i, j := 0, 0; j < n; j++ {
		acc *= nums[j]
		for i <= j && acc >= k {
			acc /= nums[i]
			i++
		}
		res += j - i + 1
	}
	return res
}
